tests/protocol/request_response.rs

```diff
@@ -314,7 +314,7 @@ async fn reject_request(transport1: Transport, transport2: Transport) {
         RequestResponseEvent::RequestFailed {
             peer: peer2,
             request_id,
-            error: RequestResponseError::Rejected
+            error: RequestResponseError::Rejected(RejectReason::ConnectionClosed)
```

Comment:
Is it actually `ConnectionClosed`, or "SubstreamClosed"?

Reply:
That makes sense! Have added a new variant `SubstreamClosed`, the `ConectionClosed` should now be constructed only when the peer disconnects to distinguish between dropping the substream and completely closing the connection

I've also added PartialEq on a few other errors, since we are most likely comparing ErrorKind's anyway :D Thanks for the review 🙏